File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/RestartOptionValue.java
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues;

import com.google.common.collect.ImmutableSet;

import java.util.Set;

/**
 * This validator is for the config_service_restart validator.
 * <p></p>
 * This seems to use the values referenced in:
 *<p></p>
 * service.c ~ line 3978
 *
 */
public class RestartOptionValue implements OptionValueInformation {
  
  private static final Set<String> validOptions
    = ImmutableSet.of("no", "on-success", "on-failure", "on-abnormal", "on-watchdog", "on-abort", "always");
  
  
  @Override
  public Set<String> getAutoCompleteOptions() {
    return validOptions;
  }
  
  @Override
  public String getErrorMessage(String value) {
  
    if (validOptions.contains(value)) {
      return null;
    } else {
      return "Expected value " + value + " does not match one of the expected options: " + validOptions;
    }
  }
  
  @Override
  public String getValidatorName() {
    return "config_parse_service_restart";
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/RestartOptionValue.kt
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues

import com.google.common.collect.ImmutableSet
import com.intellij.openapi.project.Project
import net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.Validator

/**
 * This validator is for the config_service_restart validator.
 *
 *
 * This seems to use the values referenced in:
 *
 *
 * service.c ~ line 3978
 *
 */
class RestartOptionValue : AbstractEnumOptionValue(validOptions, VALIDATOR_NAME) {

  companion object {
    private val validOptions: Set<String> = ImmutableSet.of("no", "on-success", "on-failure", "on-abnormal", "on-watchdog", "on-abort", "always")
    const val VALIDATOR_NAME = "config_parse_service_restart"

    val validators = mapOf(Validator(VALIDATOR_NAME, "0") to RestartOptionValue())

  }
}
